examples/gno.land/p/demo/grc/grc721/basic_nft.gno

```diff
@@ -69,6 +69,26 @@ func (s *basicNFT) TokenURI(tid TokenID) (string, error) {
 	return uri.(string), nil
 }
 
+func (s *basicNFT) SetTokenURI(tid TokenID, tURI TokenURI) (bool, error) {
+	// check for invalid TokenID
+	owner, found := s.owners.Get(string(tid))
+	if !found {
+		return false, ErrInvalidTokenId
+	}
+	
+	// check for the right owner
+	owner, err := s.OwnerOf(tid)
+	if err != nil {
+		return false, ErrCallerIsNotOwner
+	}
+	s.tokenURIs.Set(string(tid), string(tURI))
+	storedURI, added := s.tokenURIs.Get(string(tid))
+	if !added {
+		return false, ErrTokenURINotSet
+	}
+	return true, nil
```

Comment:
Redundant; trust that avl works correctly. (If it doesn't, we have larger problems.)

```suggestion
	s.tokenURIs.Set(string(tid), string(tURI))
	return true, nil
```